### PR TITLE
POS UI Ext: add dialog and camera scanner examples to docs

### PR DESCRIPTION
**Pull request by vctrchu**
Resolves https://github.com/Shopify/pos-next-react-native/issues/36302

### Background

Adds `CameraScanner` and `Dialog` examples and descriptions for props. I also updated some scanner examples that were using the incorrect syntax.

### 🎩
- [Dialog](https://shopify-dev.ui-extensions-psgn.victor-chu.us.spin.dev/docs/api/pos-ui-extensions/unstable/components/dialog)
- [CameraScanner](https://shopify-dev.ui-extensions-psgn.victor-chu.us.spin.dev/docs/api/pos-ui-extensions/unstable/components/camerascanner)

### CODE REVIEW

File: packages/ui-extensions/docs/surfaces/point-of-sale/reference/components/CameraScanner.doc.ts
Comment (vctrchu) [inline packages/ui-extensions/docs/surfaces/point-of-sale/reference/components/CameraScanner.doc.ts:1]: Hmmm CameraScannerProps are being weird. Will fix this tomorrow.

File: packages/ui-extensions/src/surfaces/point-of-sale/components/Dialog/Dialog.ts [hunk 1]
```diff
@@ -1,16 +1,67 @@
...
+ * - "alert": The bar at top of dialog appears yellow. Use to notify merchants about anticipated issues or unintended consequences.
+ * - "error": The bar at top of dialog appears red. Use to communicate critical issues that the merchant needs to resolve right away.
+ * - "destructive": The primary button text appears red. Use for actions that will delete information, or actions that aren’t easily reversible.
+ */
 export type DialogType = 'default' | 'alert' | 'error' | 'destructive';
 
 export interface DialogProps {
+  /**
+   * The text displayed in the title of the dialog.
+   * @param title - The text displayed in the title of the dialog.
```
Comment (vctrchu) [inline packages/ui-extensions/src/surfaces/point-of-sale/components/Dialog/Dialog.ts:15]:
```suggestion
   * @property title - The text displayed in the title of the dialog.
```

**Review by js-goupil — COMMENTED** [review]
Looks good, but I think we might still need to have the `@property` tags on the outside of the interface for it to work properly when we hover over the interface in code

File: packages/ui-extensions/docs/surfaces/point-of-sale/screenshots/camera-scanner-best-practice.png
Comment (merkoyep) [inline packages/ui-extensions/docs/surfaces/point-of-sale/screenshots/camera-scanner-best-practice.png:1]: NIT but It's difficult to see the 'Camera Scanning Error' and 'Camera scanning success' labels in the image on the web, they're super small and I can't seem to zoom in.

**Review by merkoyep — APPROVED** [review]
Looks great! I like how you used console.logs to show actions being pressed. Going to use that for my next examples!

**Pull request merged**